### PR TITLE
Check node before placement, add recipe, lower anchor placement

### PR DESCRIPTION
**Pull request by ghost**
I noticed a few things when reviewing the mod for use in my sub game. 

The first and most important is that checks have been added to ensure an anchor is only placed if the location where the anchor will be placed is an air node. Without that the player can override blocks that aren't full blocks, such as hoppers or flowers, by standing in them and using the anchor. 

Secondly I added a recipe so the anchor can be obtained without creative mode. This isn't as big of a deal because most players know how to add this themselves, but some new players may not. The recipe logic is that wool is sort of fluffy and floaty like clouds while the steel ingot is like a boat anchor. Hence floating anchor. xD It's the only thing I could think of. If you don't like the recipe and want changed let me know.

Thirdly is that I lowered the anchor placement so it is placed under the player instead of inside of them. This is more of a personal preference as it allows the player to use the anchor to catch themselves while falling. This is also more faithful to how the original angel blocks work. The reason the code subtracts 0.8 instead of 1 from the y coordinate is due to the player being able to sneak and sneaking causes the player's y coordinate to be lower. Subtracting 1 made the block placement 2 blocks lower in some situations. Using 0.8 makes it more consistent at placing it 1 block below the player. If you don't like this change I can remove it and use the changes for my sub game only.

There was also a small oversight on my part when adding the changes. Because the location is checked for an air node, checking the node for player protection isn't needed. I fixed that in a commit. You can ignore it.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -15,13 +15,18 @@ local function do_anchor_place(itemstack, player, pointed_thing)
 	
 	-- Figure out where to put the anchor
 	local place_position = player:getpos()
+	place_position.y = place_position.y - 0.8
 	log_message("action", string.format("(%d, %d, %d)", place_position.x, place_position.y, place_position.z))
 	
 	-- Place the floating anchor in the world
-	minetest.set_node(place_position, { name = "floating_anchor:floating_anchor" })
+	local player_name = player:get_player_name()
```
Comment (sbrl) [inline init.lua]: This variable doesn't appear to be used elsewhere?

**Review by sbrl — COMMENTED** [review]
Hey, thanks for the pull request! I'm honoured that you're considering including my mod in your sub-game.

I've left a comment below on a variable I'm unsure on.

Ah, thanks for that check! I was going to do that myself actually - I just hadn't gotten around to it yet, and you beat me to it :P

I don't actually play any survival at the moment, so I trust that recipe seems fair! Your recipe does seem appropriate :smiley_cat: We can always change it later if it proves to be too cheap / expensive.

Ah, awesome! Thanks for fixing that issue. Yeah, it should be below the player's feet, shouldn't it?

I'm curious though, what's player protection? Is that related to ShadowNinja's protection mod or something?

Other than my one comment below, this looks good to go! Thanks for your hard work, and keep me updated on your sub-game :smiley_cat:

**Comment by ghost** [comment]
Good catch on the variable. That was from something leftover from the protection check I removed. I removed it now. I'm not an expert on the protection in Minetest, but the way I understand it is a node can have the metadata string "owner" set to a players name. When you use minetest.is_protected(position, player_name) it will check if the player name matches that owner string to determine if they are the owner of that block. It returns true if it is protected (meaning the player name did not match) and false if the name matched.  With that being said I have yet to find where that owner string is set for the furnace in the default game. It uses it to determine if a player can add or remove items from a furnace. It's usually only used for important blocks such as chests, furnaces, other storage. I don't play multiplayer so my experience with how it works is limited.

**Comment by sbrl** [comment]
Ah, I see. Thanks for the explanation, and for removing that extra rogue line.

I've merged it in. Thanks again! :smiley_cat:

**Comment by sbrl** [comment]
Released v0.2 (and v0.3 updating the README & fixing a teensy bug in the crafting recipe)

**Pull request merged**